### ISSUE TITLE
As a front-end user, when I click on a link that takes me off of the IoGT site, a pop-up may appear that alerts me that I'm leaving a free-data environment.

### ISSUE THREAD

**Issue by lauriemarkle** [unicef/iogt#4]
When users accessing IoGT on a whitelisted IP address click on links to leave the site, such as a link to a WHO article, they should receive a pop up alert that they may incur data charges. In the CMS, administrators should be able to toggle this feature off and on. This pop-up language should be included in the multi-language support of the site. 
![Screenshot_20210218-202312](https://user-images.githubusercontent.com/74552270/108395788-3695a580-7227-11eb-8fea-241d606057e4.png)

**Comment by dannyparsons** [comment]
@isedwards Did you manage to investigate how this is done in v1 sites so that we can determine how to replicate this feature?

**Comment by isedwards** [comment]
I was assuming we would rewrite urls dynamically using Django [middleware](https://docs.djangoproject.com/en/3.2/topics/http/middleware/), but it’s possible there may be a better way to approach this. I’ll take a look at the previous implementation later today.

**Comment by lauriemarkle** [comment]
The current model is an image file which cannot be translated.

**Comment by isedwards** [comment]
Laurie - could you give some examples from the current Kenya site where you have created links to external sites?  Are the external links always created as part of a banner or are there other places where you include external links?

**Comment by dannyparsons** [comment]
Here's two examples:
- https://sa.goodinternet.org/sections/who-health-worker-covax-Text/module-2b/ - text link "Click here to learn more about EUL" to an external site
- https://zw.goodinternet.org/sections/youth/act-climate/climate-change-chatbot/ - text link "Here" to go to WhatsApp

I noticed that the "Share on Twitter" button does not go through the same alert message.

**Comment by mikowhy** [comment]
PR -> https://github.com/unicef/iogt/pull/92

**Comment by dannyparsons** [comment]
@isedwards We discovered that some external links are not being captured by the approach implemented in #92. It seems to work well for embedded links in a Paragraph block but doesn't work in some other places such as a menu item. e.g. see the UNICEF menu item here http://iogt.site/en/ which was defined as a Main Menu item as below. This should also be detected if external links are used on Flat Menus, there may be other places where external links can be defined which need detecting.

Can this be corrected easily?

![image](https://user-images.githubusercontent.com/11214914/123467751-c2aa4600-d5e8-11eb-9fec-8e748754797a.png)

**Comment by isedwards** [comment]
It's an issue with the way that this has been implemented - we're also going to have problems with migrated content from v1 because the current implementation rewrites links once at creation time (whereas v1 content was created earlier) not at run time.

When I saw the implementation I thought it may be sufficient (and also faster than a run time version), but it will only work with new content created through rich text fields.

The original suggestion on this issue was to use middleware to rewrite links dynamically.
